### PR TITLE
fix: update inherited roles in permissions page

### LINKED ISSUE #2785: [Bug] The permissions page does not take "inherited" roles into account for users who have specific permissions
## Steps to reproduce

1. Assign payer permissions to a user in a sub domain
2. Assign admin permissions to the same user in the root domain
3. Navigate to the permissions page and select the sub domain in the team filter

## Expected behaviour

* The user should show under the `admin` category, whilst they have been explicitly assigned `payer` permissions, they will have inherited the other permissions required to perform `admin` actions.

## Actual behaviour

* The user will show under the `payer` category.

## Suggestions (optional)

The "inherited" pill could show exactly which permissions have been inherited, in the example above only the `Architecture` role would be inherited. This is probably more relevant when singular custom permissions have been assigned in the sub-domain and the root domain.

### PR DESCRIPTION
**Pull request by Nortsova**
## Description
1. Assign payer permissions to a user in a sub domain
2. Assign admin permissions to the same user in the root domain
3. Navigate to the permissions page and select the sub domain in the team filter

## Testing
* Step 1. Assign Admin permissions to **fry** in Andromeda
* Step 2. Assign Payer permissions to **fry** in General
* Step 3. Assign Payer permissions to **amy** in Andromeda
* Step 4. Assign Admin permissions to **amy** in General 
* Step 5. Navigate to the permissions page to check **fry's** and **amy's** roles in Andromeda
* Step 6. **fry** should be an admin, **amy** should be an admin with _inherited_ pill
image

* Step 7. Install multi-sig extension
* Step 8. Assign Admin permissions to **amy** in Serenity  
* Step 9. Assign Payer permissions to **amy** in General
* Step 10. Assign Payer permissions to **fry** in Serenity 
* Step 11. Assign Admin permissions to **fry** in General 
* Step 12. Navigate to the permissions page to check **fry's** and **amy's** roles in Andromeda
* Step 13. **amy** should be an admin, **fry** should be an admin with _inherited_ pill
image

## TODO
Investigate the inheritance of roles and permissions in more detail.  Sync with the product team to ensure consistent behavior for inherited roles, especially in light of the upcoming implementation of nested roles. Additionally, consider the following bugs that need to be addressed: https://github.com/JoinColony/colonyCDapp/issues/2768  and https://github.com/JoinColony/colonyCDapp/issues/2967. The approach for handling inherited roles should be unified across the entire app, including the Members page (user tooltip), Action Form autofill, Permissions page, and when a user attempts specific actions.

Moreover, the inheritance for custom roles should be carefully designed.

Resolves  #2785

### CODE REVIEW

File: src/components/frame/v5/pages/MembersPage/utils.ts [hunk 1]
```diff
@@ -1,86 +1,110 @@
...
+    isMultiSig,
+  });
+
+  const inheritedPermissions = getInheritedPermissions({
+    parentPermissions,
+    currentPermissions: currentTeamPermissions,
+    isRootDomain,
+  });
+
+  // inheritedPermissions contains only a difference between parrent and current permissions
```
Comment (rumzledz) [inline src/components/frame/v5/pages/MembersPage/utils.ts:55]: Super nitpick! It should be "...between parent and..."
Thank you for this comment, much easier to digest 👌

**Review by rumzledz — APPROVED** [review]
Awesome stuff @Nortsova ✨ By following your steps, it seems that it all works as expected 👌 

Just a sidenote that it might help to clarify steps 8 - 11 to say "Assign Multi-sig {role} permissions...* just to help differentiate. Basically just add "Multi-sig" 😄 

**Individual authority permissions**


**Multi-sig Authority permissions**

**Review by bassgeta — APPROVED** [review]
This is so simple and straightforward, I love it :D Thanks for unifying the helpers and removing the unnecessary obfuscation with the `AvailablePermissions` type!
image
Both ordinary and inherited admins show up on the permissions page
With filters working too :heavy_check_mark: :
image

Both turn up as admins in `Serenity` via multisig (ignore `leela` since I've been doing some shennanigans with roles beforehand :D)
image

**Review by iamsamgibbs — APPROVED** [review]
Great work on this! Tidies it up very nicely, code all looks good.

Screenshot 2024-08-22 at 12 30 45

**Pull request merged**